deterministic-instantsend.md

```diff
@@ -0,0 +1,112 @@
+   message
+2. Calculate the RequestID from data in the `islock` message by calculating `SHA256("islock",
+   inputCount, prevTxHash1, prevTxOut1, prevTxHash2, prevTxOut2, ...)`
+3. For each LLMQ of this quorum cycle’s set, calculate `SHA256(quorumType, quorumHash, requestId)`
+4. Sort the list of LLMQs based on the result of step 3 in ascending order
+5. Use the first entry of the sorted list as the responsible LLMQ
+6. Create the SignID by calculating `SHA256(quorumHash, requestId, SHA256(version, txHash))`
+7. Use the public key of the responsible LLMQ and verify the signature against the SignID
+
+Nodes receiving `islock` messages should verify them by using the above steps. Only `ISDLOCK`
+messages with valid signatures should be propagated further using the inventory system.
+
+## The new ISDLock message
+
+When a masternode receives a recovered signature for a signing request in the quorum where it is
+active, it should use the signature to create a new p2p message, which is the `ISDLOCK` message. To
+figure out the cycle hash, take the quorum hash corresponding to the LLMQ that created the recovered
+signature. Then find the last known block where this quorum was responsible for signing the `islock`
+message. Often this is the current block, but in rare situations it might be a prior one if quorums
+have just cycled. From this last known block, find the first block of that cycle. This is the last
+block before quorums changed for the quorum type used for `islock` messages.
+
+It is possible that a quorum is active before and after quorum cycling. It is also possible that the
+quorum responsible for a signing request before and after cycling is the same. This could lead to
+the creation of two `islock` messages, distinct only by the fact that their cycle hash is different.
+
+The new message has the following structure (fields in bold are not present in the previously used
+`islock` message):
+
+| Field | Type | Size | Description |
+|-|-|-|-|
+| **version** | uint8 | 1 |  The version of the islock message |
+| inputCount | compactSize uint | 1 - 9 | Number of inputs in the transaction |
+| inputs | COutpoint[] | `inputCount` * 36 | Inputs of the transaction. COutpoint is a uint256 (hash of previous transaction) and a uint32 (output index) |
+| txid | uint256 | 32 | Transaction id (hash of the transaction) |
+| **cycleHash** | uint256 | 32 | Block hash of first block of the cycle in which the quorum signing this `islock` is active |
+| sig | BLSSig | 96 | Recovered signature from the signing request/session |
+
+### **Choosing the active LLMQ to perform signing**
+
+Choosing the active LLMQ to perform signing should follow the same steps as defined in [DIP-0007 -
+Choosing the active LLMQ to perform
+signing](https://github.com/dashpay/dips/blob/master/dip-0007.md#choosing-the-active-llmq-to-perform-signing).
+
+| Field | Type | Size | Description |
```

Comment:
@QuantumExplorer Is this table supposed to be here? I'm not sure I understand its purpose.

Reply:
Removed after talking with Pasta